tests/end2end/model.py

```diff
@@ -63,6 +75,13 @@ def low_rank_update(t, rank):
     return {"A": R, "B": C}
 
 
+def make_ia3_update(value):
+    ia3 = np.random.randn(*value.shape)
```

Comment:
You need to add `.astype(np.float32)` here, right now these are float64 so when the final diff is done the git repo has things saved as float32 but the file on disk has them saved as float64